common.gypi

```diff
@@ -279,13 +285,29 @@
           '_target_name!="<(node_core_target_name)")', {
         'cflags!': ['-Werror'],
       }],
+      # TODO(targos): Remove condition and always use LanguageStandard options
+      # once node-gyp supports them.
+      ['clang==1', {
+        'msvs_settings': {
+          'VCCLCompilerTool': {
+            'LanguageStandard': 'stdcpp20',
+            'LanguageStandard_C': 'stdc11',
+          },
+        },
+      }, {
+        'msvs_settings': {
+          'VCCLCompilerTool': {
+            'AdditionalOptions': [
+              '-std:c++20',
+            ],
+          },
+        },
```

Comment:
Have I made a mistake here? The option is not applied.

Reply:
In the old PR, I had verified that the following would work...

```
        'LanguageStandard': 'stdcpp20',
         'LanguageStandard_C': 'stdc11',
```

I don't think that `-std:c++20` is an expected MS syntax, though maybe it can sometimes work.

Reply:
I used that when `clang==1`. We cannot use that unconditionally in `common.gypi` because it would break users of `node-gyp` which doesn't have the gyp changes yet.

Reply:
Actually, there's something that prevents the all the conditions from being evaluated (even the existing one `'error_on_warn=="false"'`). If I add a syntax error to it, nothing changes.